### ISSUE TITLE
It looks like Collection Expressions feature is missing tests for missing APIs that are required by `SynthesizedInlineArrayAsReadOnlySpanMethod`

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#71160]
In fact, it looks like the checks aren't actually proactively performed. Here is why I think this is the case. If in ```CollectionBuilder_02A``` unit-test I make the following change:
```diff
 comp = CreateCompilation(new[] { sourceB, s_collectionExtensions }, references: new[] { refA }, targetFramework: targetFramework, options: TestOptions.ReleaseExe);
+comp.MakeMemberMissing(WellKnownMember.System_Runtime_InteropServices_MemoryMarshal__CreateReadOnlySpan);
 comp.VerifyEmitDiagnostics();
```

and then run the test. The test fails with:
```
Expected:
Actual:
// error CS0656: Missing compiler required member 'System.Runtime.InteropServices.MemoryMarshal.CreateReadOnlySpan'
// 
Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "").WithArguments("System.Runtime.InteropServices.MemoryMarshal", "CreateReadOnlySpan").WithLocation(1, 1)
```
on the ``` comp.VerifyEmitDiagnostics();``` line. Note that the error is reported without any reference to the source code.

The compiler doesn't crash because ```SynthesizedInlineArrayAsReadOnlySpanMethod.GenerateMethodBody``` catches an exception and converts it to a diagnostic. However, this is just a stop gap try/catch, it was never meant to be the way to check and report missing APIs.

Other "InlineArray" synthesized helpers used by "Collection Expressions" feature likely have similar issues.

**Comment by AlekseyTs** [comment]
@cston FYI
